purchase_quick/models/product_product.py

```diff
@@ -0,0 +1,119 @@
+# coding: utf-8
+# © 2014 Today Akretion
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# @author Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models, _
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    qty_to_purchase = fields.Float(
+        compute='_compute_purchase_qty',
+        inverse='_inverse_set_purchase_qty',
+        oldname='purchase_qty',
+        help="Set this quantity to create a new purchase line "
+             "for this product or update the existing one."
+    )
+    po_line_ids = fields.One2many(
+        comodel_name='purchase.order.line', inverse_name='product_id',
+        help='Technical: used to compute quantities to purchase.',
+    )
+
+    def _prepare_purchase_line(self, purchase):
+        return {
+            'product_id': self.id,
+            'product_qty': self.qty_to_purchase,
+            'order_id': purchase.id,
+        }
+
+    def _get_purchase_line(self, purchase):
+        return self.env['purchase.order.line'].search([
+            ('product_id', '=', self.id),
+            ('order_id', '=', purchase.id),
+        ], limit=1)
+
+    def _add_purchase_line(self, purchase):
+        vals = self._prepare_purchase_line(purchase)
+        line = self.env['purchase.order.line'].new(vals)
+        vals = self._complete_purchase_line_vals(line, vals)
+        if not vals.get('price_unit'):
+            vals['price_unit'] = 0.0
+        self.env['purchase.order.line'].create(vals)
+
+    def _update_purchase_line(self, purchase_line):
+        if self.qty_to_purchase:
+            # apply the on change to update price unit if depends on qty
+            vals = {'product_qty': self.qty_to_purchase}
+            self._complete_purchase_line_vals(purchase_line, vals)
+            purchase_line.write(vals)
+        else:
+            purchase_line.unlink()
+
+    def _complete_purchase_line_vals(self, line, vals):
+        vals = line.play_onchanges(vals, ['product_id'])
+        if vals.get('taxes_id'):
+            vals['taxes_id'] = [(6, 0, vals['taxes_id'])]
+        return vals
+
+    def _inverse_set_purchase_qty(self):
+        purchase = self.env['purchase.order'].browse(
+            self._context.get('purchase_id'))
```

Comment:
please use self.env.context

Reply:
@bealdav It's not wrong Or not blocking anywhere but @PierrickBrun It's Improvements. :smile: